crates/biome_js_semantic/src/events.rs

```diff
@@ -321,8 +324,11 @@ impl SemanticEventExtractor {
                     false,
                 );
             }
-
-            _ => {}
+            _ => {
+                if let Some(node) = node.clone().cast::<AnyTsType>() {
```

Comment:
We could clone only when the cast is possible:
```suggestion
                if let Some(node) = AnyTsType::cast_ref(node) {
```